corgidrp/data.py

```diff
@@ -203,8 +205,8 @@ def __init__(self, data_or_filepath, pri_hdr=None, ext_hdr=None, err = None, dq
             self.filedir = "."
             self.filename = ""
             if err is not None:
-                if np.shape(self.data) != np.shape(err):
-                    raise ValueError("The shape of err is {0} while we are expecting shape {1}".format(err.shape, self.data.shape))
+                if np.shape(self.data) != np.shape(err)[-2:]:
+                    raise ValueError("The shape of err is {0} while we are expecting shape {1}".format(err.shape[-2:], self.data.shape))
                 self.err = err
```

Comment:
I think we should just stick with 3D arrays. 
When it's first initialized it's ok for it to be [1,npix,npix] rather than [npix,npix]

Reply:
The question 2 or 3-D, I think now both is possible, but I find it a bit awkward as a users to generate a 3D error array on a 2D signal array to initialize the image. In my primitive imagination the err parameter is only used while initializing the image class (if you have an error already), otherwise the add_error function should be used or the fits hdu is used.

Reply:
Could the code here take a 2-D error array as input, and change it into a 3-D array when creating the Image class? That way, we can conveniently pass in 2-D arrays, but the Image class will always maintain a 3-D error array format for simplicity.